test/unit/OracleDDLHelper.spec.ts

```diff
@@ -1,23 +1,105 @@
 
 import 'reflect-metadata';
 import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
 
 import OracleDDLHelper from '../../source/DataAccess/Oracle/OracleDDLHelper';
+import IConnectionPool from '../../source/DataAccess/IConnectionPool';
 
 const expect = chai.expect;
 
 describe('OracleDDLHelper', () => {
 
+    const createTestContext = (tableAssociations: Array<[string, string]> = []) => {
```

Comment:
I got fed up with the caveats that come with trying to stub layers of stubbed objects and just went with anonymous stubs.